app/routes/analytics.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express')
+const router = express.Router()
+const Counter = require('../models/routeCounter')
+
+const DENYLIST_URLS = [
+    '/articles/approve',
+    '/login',
+    '/register',
+    '/logout',
+    '/image',
+    '/articles/image'
+]
+
+router.post('/fingerprint', async (req, res) => {
+    const {currentUrl, fingerprint} = req.body
+    if(!currentUrl || !fingerprint || !isBase64(fingerprint)) return res.sendStatus(422) //Unprocessable Entity aka bad parameters
+    if(DENYLIST_URLS.find(BAD_URL => currentUrl.includes(BAD_URL))) return res.sendStatus(200) //Don't store these route visits
+    try {
```

Comment:
Could u restrict try block to only the async code in order to avoid unnecessary intention. This code looks quite clumsy

Reply:
Sure